fastmri_examples/adaptive_varnet/eval_pretrained_adaptive_varnet.py

```diff
@@ -0,0 +1,254 @@
+import numpy as np
+import pytorch_lightning as pl
+import torch
+from fastmri import evaluate
+from fastmri.data.mri_data import fetch_dir
+from fastmri.data.transforms import MiniCoilTransform
+from fastmri.pl_modules import ActiveVarNetModule, FastMriDataModule
+from subsample import create_mask_for_mask_type
+
+
+def str2bool(v):
+    if isinstance(v, bool):
+        return v
+    if v.lower() in ("yes", "true", "t", "y", "1"):
+        return True
+    elif v.lower() in ("no", "false", "f", "n", "0"):
+        return False
+    else:
+        raise ValueError("Boolean value expected.")
+
+
+def entropy(prob_mask: torch.Tensor):
+    ent = -(prob_mask * prob_mask.log() + (1 - prob_mask) * (1 - prob_mask).log())
+    ent[prob_mask == 0] = 0
+    ent[prob_mask == 1] = 0
+    return ent
+
+
+def save_images(recon):
+    for i, tensor in enumerate(recon):
+        plt.imshow(tensor.detach().cpu().numpy(), cmap="gist_gray")
+        plt.colorbar()
+        plt.savefig(f"/scratch/lep/active_acq/recon_{i}.png")
+        plt.close()
+
+
+def cli_main(args):
+    pl.seed_everything(0)
+
+    # ------------
+    # data
+    # ------------
+    # this creates a k-space mask for transforming input data
+    mask = create_mask_for_mask_type(
+        args.mask_type,
+        args.center_fractions,
+        args.accelerations,
+        args.skip_low_freqs,
+    )
+
+    val_transform = MiniCoilTransform(
+        mask_func=mask,
+        num_compressed_coils=4,
+        crop_size=(128, 128),
+    )
+
+    data_module = FastMriDataModule(
+        data_path=args.data_path,
+        challenge=args.challenge,
+        train_transform=val_transform,
+        val_transform=val_transform,
+        test_transform=val_transform,
+        test_split=args.test_split,
+        test_path=args.test_path,
+        sample_rate=args.sample_rate,
+        volume_sample_rate=args.volume_sample_rate,
+        batch_size=args.batch_size,
+        num_workers=args.num_workers,
+    )
+
+    # ------------
+    # model
+    # ------------
+    device = "cuda:0"
+    model = ActiveVarNetModule.load_from_checkpoint(args.load_checkpoint)
+    model.to(device)
+
+    ssim_loss = fastmri.SSIMLoss()
+
+    all_prob_masks = []
+    all_ssims, all_psnrs, all_nmses = [], [], []
+    data_loader = (
+        data_module.val_dataloader()
+        if args.data_mode == "val"
+        else data_module.train_dataloader()
+    )
+    with torch.no_grad():
+        for i, batch in enumerate(data_loader):
+            if i == args.num_batches:
+                break
+            kspace, masked_kspace, mask, target, fname, slice_num, max_value, _ = batch
+
+            # import pdb
+            # pdb.set_trace()
+            output, extra_outputs = model(
+                kspace.to(device),
+                masked_kspace.to(device),
+                mask.to(device),
+                fname,
+                slice_num,
+            )
+            prob_masks_list = extra_outputs["prob_masks"]
```

Comment:
This or lines below will probably give an error for Equispaced models (no prob_mask). We already have an `if all_prob_masks` check in line 156, so we should add a similar check here (`if "prob_masks" in extra_output:`).

Reply:
I think in its current version this script doesn't support Equispaced models at all. Is there a way to automatically detect the type of model from the checkpoint? In my internal scripts I used to do it from the wandb id, but I don't think we should rely on that here.

Reply:
Aah yes, good point. If we want to make this work for equispaced models we probably have to do a try-except on the checkpoint loading with ActiveVarNetModule and VarNetModule. In the latter case we'd also need to check if mask type is `equispaced` or `random` if we want to be complete.

The eval script also assumes (128, 128) crop and 4 compressed coils, whereas the train scripts leaves this as an argument (and if no coil-compression is used, it uses VarNetDataTransform instead: https://github.com/facebookresearch/fastMRI/blob/mmuckley/adaptive_varnet_updates/fastmri_examples/adaptive_varnet/train_adaptive_varnet_demo.py#L410 ).

I don't really see a problem with leaving the eval script as is, but these changes would make it more compatible with the full optionality that the training script provides.